### PR TITLE
Correct

### PR DESCRIPTION
**Pull request by VladislavBrykov**

### CODE REVIEW

File: Api-Router/Users/user.controller.ts [hunk 1]
```diff
@@ -1,73 +1,44 @@
...
     const { password, phoneEmail } = req.body;
-    const resRegistration: boolean = await resultRegistration(phoneEmail, password);
-
-    if (resRegistration) {
-        return res.status(200).json({ status: "registration successful" });
-    } else {
-        return res.status(404).json({ status: "registration error^ user exists" });
-    }
+    const resRegistration: boolean = await userControllerData.serviceRegistration(phoneEmail, password);
+    return (resRegistration ? res.status(200).json({ status: 'registration successful' }) : res.status(404).json({ status: 'registration error, user exists' }))
```
Comment (roman1bondarenko) [inline Api-Router/Users/user.controller.ts:14]: return resRegistration ? 
       res.status(200).json({ status: 'registration successful' }) : 
       res.status(404).json({ status: 'registration error, user exists' })

File: Api-Router/Users/user.controller.ts [hunk 2]
```diff
@@ -1,73 +1,44 @@
...
-
-    const resLogout: boolean = await resultLogout(token, all)
-
-    if (resLogout) {
-        res.status(200).json({ status: true });
-    } else {
-        res.status(200).json({ status: "token error" });
-    }
+    let all = true;
+    req.url.indexOf('true') >= 0 ? all : all = null
```
Comment (roman1bondarenko) [inline Api-Router/Users/user.controller.ts:32]: express query params

Comment (roman1bondarenko) [reply Api-Router/Users/user.controller.ts:32]: PLUS check express core and features!!!!
http://expressjs.com/en/4x/api.html#req.params?foo=qwertyu

File: Service/Users/users.servece.data.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        await User.create(registrationUser);
+        return true
+    }
+}
+
+async function serviceLogout(token: string, all: boolean) {
+    const searchUser = await UserDevice.findOne({ where: { token: token } })
+
+    if (searchUser) {
+        all ? UserDevice.update({ token: null }, { where: { token: token } }) : User.update({ token: null }, { where: { phoneEmail: searchUser.phoneEmail } })
```
Comment (roman1bondarenko) [inline Service/Users/users.servece.data.ts:46]: all ? 
         UserDevice.update({ token: null }, { where: { token: token } }) : 
         User.update({ token: null }, { where: { phoneEmail: searchUser.phoneEmail } });


check DB models!!!

File: Service/Users/utils/user.service.helpers.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    if (searchUser) {
+        function updateObject(nameObject) {
+            nameObject.update({ token: newToken }, {
+                where: {
+                    phoneEmail: data.phoneEmail
+                }
+            });
+        }
+        await updateObject(searchUser);
+        await updateObject(UserDevice);
```
Comment (roman1bondarenko) [inline Service/Users/utils/user.service.helpers.ts:20]: Promise.all();

File: Service/Users/utils/user.service.helpers.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+async function searchUserService(token: string) {
+    const searchUser = await UserDevice.findOne({ where: { token: token } })
+    return (searchUser ? searchUser : false)
+}
+
+async function searchUserTable(phoneEmail: string) {
+    const searchUser = await User.findOne({ where: { phoneEmail: phoneEmail } })
+    return (searchUser ? searchUser : false)
+}
+
```
Comment (roman1bondarenko) [inline Service/Users/utils/user.service.helpers.ts:35]: use nore generic function and check again DB models and structure :)

**Review by roman1bondarenko — CHANGES_REQUESTED** [review]

File: Service/Users/utils/user.service.helpers.ts (hunk 5 above)
Comment (VladislavBrykov) [reply Service/Users/utils/user.service.helpers.ts:35]: I use table User, because in this table save params typeId. Structure is correct

**Pull request merged**